integration/aggregate_compat_test.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+// aggregateStageCompatTestCase describes aggregate compatibility test case.
+type aggregateStageCompatTestCase struct {
+	skip       string                   // skip test for all handlers, must have issue number mentioned
+	pipeline   bson.A                   // required, unspecified $sort appends bson.D{{"$sort", bson.D{{"_id", 1}}}}
+	resultType compatTestCaseResultType // defaults to nonEmptyResult
+}
+
+// testAggregateStageCompat tests aggregate stage compatibility test cases.
+func testAggregateStageCompat(t *testing.T, testCases map[string]aggregateStageCompatTestCase) {
```

Comment:
If we use this helper for in tests with single stages but also multiple stages, then I would suggest to make it plural
```suggestion
func testAggregateStagesCompat(t *testing.T, testCases map[string]aggregateStageCompatTestCase) {
```

Reply:
Thanks, let's do that.